txpipe/twopoint_fourier.py

```diff
@@ -546,12 +547,33 @@ def compute_power_spectra(self, pixel_scheme, i, j, k, maps, workspaces, ell_bin
 
         workspace = workspaces[(i,j,k)]
 
-        # Get the coupled noise C_ell values to give to the master algorithm
-        cl_noise = self.compute_noise(i, j, k, ell_bins, maps, workspace)
-
-        # Run the master algorithm
-        c = nmt.compute_full_master(field_i, field_j, ell_bins,
-            cl_noise=cl_noise, cl_guess=cl_guess, workspace=workspace, n_iter=1)
+        if self.config['analytic_noise']==False:
```

Comment:
I usually prefer to minimize the mental logical people have to do and just do `if self.config['analytic_noise']:` (which would mean swapping the order here).

Reply:
Done